### PR TITLE
issue #162 work in progress - CLI command reference docs

### PR DESCRIPTION
**Pull request by fozy81**
A work in progress, drafting list of all CLI commands.

### CODE REVIEW

**Comment by jenweber** [comment]
Hi, thanks for pulling this together! Can you @ me when you are ready for another review?

I think it might be good to divide each major section into its own codeblock, for readability (i.e. one for components, one for addons, etc)

File: guides/advanced-use/cli-commands-reference.md [hunk 1]
```diff
@@ -0,0 +1,279 @@
+In their daily work, most Ember developers use only a small number of CLI commands. See the [Basic use](release/basic-use/cli-commands/) section for a guide to common CLI commands and options.
+  
+This page shows a full list of Ember CLI commands and options. You can view this information at anytime by typing the following
+
+```bash
+ember --help
+```
+As Ember CLI is constantly evolving it's best to use this command to view the latest help for the version of Ember CLI you are currently using.
+
+Available commands in ember-cli (alphabetical):
```
Comment (jenweber) [inline guides/advanced-use/cli-commands-reference.md:10]:
```suggestion
Available commands in `ember-cli` (alphabetical):
```

This change will help tests pass.

File: guides/advanced-use/cli-commands-reference.md [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+  --run-codemod (Boolean) run any associated
+   codemods without prompting
+
+
+Available commands from ember-cli-template-lint:
+ember template-lint:print-failing
+  Get current list of all failing templates. 
+  This can be used to update the `pending` 
+  listing in `.template-lintrc.js`.
+```
```
Comment (jenweber) [inline guides/advanced-use/cli-commands-reference.md:279]:
```suggestion
```

```


This change will help tests pass.

**Comment by fozy81** [comment]
Thanks @jenweber for first review - now ready for a second. All checks passing this time! :fireworks: - I've split up the commands into separate sections. Looks much more readable. Let me know if there's anything else needing changing.

File: guides/advanced-use/cli-commands-reference.md [hunk 3]
```diff
@@ -0,0 +1,303 @@
+In their daily work, most Ember developers use only a small number of CLI commands. See the [Basic use](release/basic-use/cli-commands/) section for a guide to common CLI commands and options.
```
Comment (jenweber) [inline guides/advanced-use/cli-commands-reference.md:1]:
```suggestion
In their daily work, most Ember developers use only a small number of CLI commands. See the [Basic use](../basic-use/cli-commands/) section for a guide to common CLI commands and options.
```

**Review by jenweber — APPROVED** [review]
Thank you! This looks great! I added one fix for the link to the Basic Use page. They work a little bit strangely in this project.

**Pull request merged**